### PR TITLE
add transaction-atomicity parameter to TiCDC

### PR DESCRIPTION
**Pull request by CharlesCheung96**
### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### What is changed, added or deleted? (Required)
add transaction-atomicity parameter to TiCDC

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [x] v6.2 (TiDB 6.2 versions)
- [x] v6.1 (TiDB 6.1 versions) **needs version specific change**
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?
ref: https://github.com/pingcap/tiflow/pull/6038

- This PR is translated from:
- Other reference link(s): https://github.com/pingcap/docs-cn/issues/11064

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Review by leoppro — APPROVED** [review]

**Comment by TomShawn** [comment]
@CharlesCheung96 Could you please sign the [CLA](https://github.com/pingcap/docs-cn/pull/10313#issuecomment-1169847705) as reminded by the [CLAassistant](https://github.com/CLAassistant)? Thanks~

**Comment by CharlesCheung96** [comment]
> @CharlesCheung96 Could you please sign the [CLA](https://github.com/pingcap/docs-cn/pull/10313#issuecomment-1169847705) as reminded by the [CLAassistant](https://github.com/CLAassistant)? Thanks~

Done, PTAL!

File: ticdc/ticdc-faq.md [hunk 1]
```diff
@@ -188,8 +188,9 @@ TiCDC 使用 PD 内部的 etcd 来存储元数据并定期更新。因为 etcd 
 
 ## TiCDC 支持同步大事务吗？有什么风险吗？
 
-TiCDC 对大事务（大小超过 5 GB）提供部分支持，根据场景不同可能存在以下风险：
+根据场景不同，TiCDC 同步大事务存在以下风险：
 
++ 可能导致主从同步延迟大幅增高
 + 当 TiCDC 内部处理能力不足时，可能出现同步任务报错 `ErrBufferReachLimit`。
 + 当 TiCDC 内部处理能力不足或 TiCDC 下游吞吐能力不足时，可能出现内存溢出 (OOM)。
```
Comment (nongfushanquan) [inline ticdc/ticdc-faq.md:196]: if errors about big transaction  happen,  enabling the splitting big transaction is perhaps a better choice

**Review by leoppro — APPROVED** [review]
LGTM

**Review by nongfushanquan — APPROVED** [review]

File: ticdc/ticdc-overview.md [hunk 2]
```diff
@@ -64,6 +64,10 @@ TiCDC 的系统架构如下图所示：
     - TiCDC 以表为单位拆分跨表事务，**不保证**跨表事务的原子性。
     - TiCDC **保证**单行的更新与上游更新顺序一致。
 
+> **注意：**
+  >
+  > 从 v6.1.1 版本起，TiCDC 可以通过配置 transaction-atomicity 控制是否拆分单表事务。拆分事务可以大幅降低 MySQL sink 同步大事务的延时和内存消耗。
```
Comment (TomShawn) [inline ticdc/ticdc-overview.md:69]:
```suggestion
> **注意：**
>
> 从 v6.1.1 版本起，你可以通过配置 sink uri 参数 `transaction-atomicity` 来控制 TiCDC 是否拆分单表事务。拆分事务可以大幅降低 MySQL sink 同步大事务的延时和内存消耗。
```

File: ticdc/ticdc-faq.md [hunk 3]
```diff
@@ -188,12 +188,15 @@ TiCDC 使用 PD 内部的 etcd 来存储元数据并定期更新。因为 etcd 
 
 ## TiCDC 支持同步大事务吗？有什么风险吗？
 
-TiCDC 对大事务（大小超过 5 GB）提供部分支持，根据场景不同可能存在以下风险：
+根据场景不同，TiCDC 同步大事务存在以下风险：
```
Comment (TomShawn) [inline ticdc/ticdc-faq.md:191]: 这里是否要说明下大事务是多大？

File: ticdc/ticdc-faq.md [hunk 4]
```diff
@@ -188,12 +188,15 @@ TiCDC 使用 PD 内部的 etcd 来存储元数据并定期更新。因为 etcd 
...
 
++ 可能导致主从同步延迟大幅增高
```
Comment (TomShawn) [inline ticdc/ticdc-faq.md:193]:
```suggestion
+ 可能导致主从同步延迟大幅增加
```

File: ticdc/ticdc-overview.md [hunk 5]
```diff
@@ -64,6 +64,10 @@ TiCDC 的系统架构如下图所示：
     - TiCDC 以表为单位拆分跨表事务，**不保证**跨表事务的原子性。
     - TiCDC **保证**单行的更新与上游更新顺序一致。
 
+> **注意：**
+  >
+  > 从 v6.2 版本起，TiCDC 可以通过配置 transaction-atomicity 控制是否拆分单表事务。拆分事务可以大幅降低 MySQL sink 同步大事务的延时和内存消耗。
```
Comment (TomShawn) [inline ticdc/ticdc-overview.md]:
```suggestion
> **注意：**
>
> 从 v6.2 版本起，你可以通过配置 sink uri 参数 `transaction-atomicity` 来控制 TiCDC 是否拆分单表事务。拆分事务可以大幅降低 MySQL sink 同步大事务的延时和内存消耗。
```

File: ticdc/ticdc-overview.md [hunk 6]
```diff
@@ -64,6 +64,10 @@ TiCDC 的系统架构如下图所示：
     - TiCDC 以表为单位拆分跨表事务，**不保证**跨表事务的原子性。
     - TiCDC **保证**单行的更新与上游更新顺序一致。
 
+> **注意：**
+  >
+  > 从 v6.2 版本起，你可以通过配置 sink uri 参数 [`transaction-atomicity` 参数](/ticdc/manage-ticdc.md#sink-uri-配置-mysqltidb) 来控制 TiCDC 是否拆分单表事务。拆分事务可以大幅降低 MySQL sink 同步大事务的延时和内存消耗。
```
Comment (TomShawn) [inline ticdc/ticdc-overview.md:69]:
```suggestion
> **注意：**
>
> 从 v6.2 版本起，你可以通过配置 sink uri 参数 [`transaction-atomicity` 参数](/ticdc/manage-ticdc.md#sink-uri-配置-mysqltidb) 来控制 TiCDC 是否拆分单表事务。拆分事务可以大幅降低 MySQL sink 同步大事务的延时和内存消耗。
```

File: ticdc/manage-ticdc.md [hunk 7]
```diff
@@ -185,6 +185,7 @@ URI 中可配置的参数如下：
 | `ssl-cert`     | 连接下游 MySQL 实例所需的证书文件路径（可选） |
 | `ssl-key`      | 连接下游 MySQL 实例所需的证书密钥文件路径（可选） |
 | `time-zone`    | 连接下游 MySQL 实例时使用的时区名称，从 v4.0.8 开始生效。（可选。如果不指定该参数，使用 TiCDC 服务进程的时区；如果指定该参数但使用空值，则表示连接 MySQL 时不指定时区，使用下游默认时区） |
+| `transaction-atomicity`      | 指定事务原子性级别（可选，默认值为`table`），当该值为`table`时 TiCDC 保证单表事务的原子性，当该值为`none`时 TiCDC 会拆分单表事务 |
```
Comment (TomShawn) [inline ticdc/manage-ticdc.md:188]:
```suggestion
| `transaction-atomicity`      | 指定事务的原子性级别（可选，默认值为 `table`）。当该值为`table` 时 TiCDC 保证单表事务的原子性，当该值为 `none` 时 TiCDC 会拆分单表事务 |
```

File: ticdc/ticdc-faq.md [hunk 8]
```diff
@@ -188,12 +188,15 @@ TiCDC 使用 PD 内部的 etcd 来存储元数据并定期更新。因为 etcd 
...
 + 当 TiCDC 内部处理能力不足时，可能出现同步任务报错 `ErrBufferReachLimit`。
 + 当 TiCDC 内部处理能力不足或 TiCDC 下游吞吐能力不足时，可能出现内存溢出 (OOM)。
 
-当遇到上述错误时，建议将包含大事务部分的增量数据通过 BR 进行增量恢复，具体操作如下：
+从 v6.2 版本开始，TiCDC 支持拆分单表事务功能，可大幅降低同步大事务的延时和内存消耗。因此，建议用户在业务对事务原子性要求不高的场景下，通过设置 [`transaction-atomicity` 参数](/ticdc/manage-ticdc.md#sink-uri-配置-mysqltidb)打开拆分事务功能以解决可能出现的同步延迟和 OOM 问题。
```
Comment (TomShawn) [inline ticdc/ticdc-faq.md:197]:
```suggestion
从 v6.2 版本开始，TiCDC 支持拆分单表事务功能，可大幅降低同步大事务的延时和内存消耗。因此，在业务对事务原子性要求不高的场景下，建议通过设置 sink uri 参数 [`transaction-atomicity`](/ticdc/manage-ticdc.md#sink-uri-配置-mysqltidb) 打开拆分事务功能以解决可能出现的同步延迟和 OOM 问题。
```

File: ticdc/ticdc-faq.md (hunk 3 above)
Comment (CharlesCheung96) [reply ticdc/ticdc-faq.md:191]: 准确讲大事务和小事务是没有严格界定的，所以感觉这里不需要给出一个具体的数值？

File: ticdc/manage-ticdc.md [hunk 9]
```diff
@@ -185,6 +185,7 @@ URI 中可配置的参数如下：
 | `ssl-cert`     | 连接下游 MySQL 实例所需的证书文件路径（可选） |
 | `ssl-key`      | 连接下游 MySQL 实例所需的证书密钥文件路径（可选） |
 | `time-zone`    | 连接下游 MySQL 实例时使用的时区名称，从 v4.0.8 开始生效。（可选。如果不指定该参数，使用 TiCDC 服务进程的时区；如果指定该参数但使用空值，则表示连接 MySQL 时不指定时区，使用下游默认时区） |
+| `transaction-atomicity`      | 指定事务的原子性级别（可选，默认值为 `table`）。当该值为`table` 时 TiCDC 保证单表事务的原子性，当该值为 `none` 时 TiCDC 会拆分单表事务 |
```
Comment (TomShawn) [inline ticdc/manage-ticdc.md:188]:
```suggestion
| `transaction-atomicity`      | 指定事务的原子性级别（可选，默认值为 `table`）。当该值为 `table` 时 TiCDC 保证单表事务的原子性，当该值为 `none` 时 TiCDC 会拆分单表事务 |
```

File: ticdc/ticdc-overview.md [hunk 10]
```diff
@@ -64,6 +64,10 @@ TiCDC 的系统架构如下图所示：
     - TiCDC 以表为单位拆分跨表事务，**不保证**跨表事务的原子性。
     - TiCDC **保证**单行的更新与上游更新顺序一致。
 
+> **注意：**
+>
+> 从 v6.2 版本起，你可以通过配置 sink uri 参数 [`transaction-atomicity` 参数](/ticdc/manage-ticdc.md#sink-uri-配置-mysqltidb) 来控制 TiCDC 是否拆分单表事务。拆分事务可以大幅降低 MySQL sink 同步大事务的延时和内存消耗。
```
Comment (TomShawn) [inline ticdc/ticdc-overview.md:69]:
```suggestion
> 从 v6.2 版本起，你可以通过配置 sink uri 参数 [`transaction-atomicity`](/ticdc/manage-ticdc.md#sink-uri-配置-mysqltidb) 来控制 TiCDC 是否拆分单表事务。拆分事务可以大幅降低 MySQL sink 同步大事务的延时和内存消耗。
```

**Review by TomShawn — APPROVED** [review]
LGTM

File: ticdc/ticdc-faq.md [hunk 11]
```diff
@@ -190,10 +190,13 @@ TiCDC 使用 PD 内部的 etcd 来存储元数据并定期更新。因为 etcd 
 
 TiCDC 对大事务（大小超过 5 GB）提供部分支持，根据场景不同可能存在以下风险：
 
++ 可能导致主从同步延迟大幅增高
```
Comment (shichun-0415) [inline ticdc/ticdc-faq.md:193]:
```suggestion
+ 可能导致主从同步延迟大幅增高。
```

**Review by shichun-0415 — APPROVED** [review]
LGTM

**Pull request merged**